src/components/DocumentsPage/index.js

```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react';
+// components
+import DocumentsEditor from './components/DocumentsEditor';
+import DocumentsManager from './components/DocumentsManager';
+// styles 
+import './styles.css';
+
+class DocumentsPage extends Component {
+  
+  constructor(props) {
+    super();
+    this.state = { documents: {}, document: null, unsaved: {} }
+  };
+
+  next_id = 0;
+  // 'database' documents
+  documents = {};
+
+  onDocumentEdit = (inputName, inputValue) => {
+    this.setState((prevState) => {
+      var document = {...prevState.document, [inputName]: inputValue, unsaved: true };
```

Comment:
You're properly updating document here, I'll check locally why editor doesn't work without spreading a prop, then come back to this